test/blockchaintest/blockchaintest_runner.cpp

```diff
@@ -112,6 +112,30 @@ bool validate_block(
 {
     // NOTE: includes only block validity unrelated to individual txs. See `apply_block`.
 
+    if (test_block.block_info.gas_used > test_block.block_info.gas_limit)
+        return false;
+
+    // Some tests have gas limit at INT64_MAX, so we cast to uint64_t to avoid overflow.
+    const auto parent_header_gas_limit_u64 = static_cast<uint64_t>(parent_header.gas_limit);
+    const auto test_block_gas_limit_u64 = static_cast<uint64_t>(test_block.block_info.gas_limit);
+    if (test_block_gas_limit_u64 >=
+        parent_header_gas_limit_u64 + parent_header_gas_limit_u64 / 1024)
+        return false;
+    if (test_block_gas_limit_u64 <=
+        parent_header_gas_limit_u64 - parent_header_gas_limit_u64 / 1024)
+        return false;
+
+    if (test_block.block_info.gas_limit < 5000)
```

Comment:
What a hell?

Reply:
It's requirement from yellow paper, also mentioned in EIP-1559
![image](https://github.com/user-attachments/assets/55b80808-082c-4bcf-8755-681971e59a3d)

But tests don't cover it, so should I remove?
cc @winsvega

Reply:
This is funny. Does it mean an empty block with 4999 gas limit is invalid?

Maybe it comes for the gas increase/decrease limits... For low values the increase may become 0 forever.

Usually in such case I use `assert()`. But fine to leave the check too (we can later add an EEST test). Just add the comment.